src/Pos/Communication/Limits.hs

```diff
@@ -94,13 +99,13 @@ instance MessageLimitedPure PVSS.ExtraGen where
     msgLenLimit = 33
 
 instance MessageLimitedPure (AbstractHash Blake2b_224 a) where
-    msgLenLimit = 28
+    msgLenLimit = 32
```

Comment:
This number was adjusted empirically after seeing the tests failing, but trying to answer it here I can speculate that's because we are calling `unsafeAbstractHash` in the `Arbitrary` instance. Such `Arbitrary` instance is defined as follows:

``` haskell
instance (HashAlgorithm algo, Bi a) =>
         ArbitraryUnsafe (AbstractHash algo a) where
    arbitraryUnsafe = unsafeAbstractHash <$>
        choose (minBound, maxBound :: Word64)
```

whereas `unsafeAbstractHash` is:

``` haskell
unsafeAbstractHash
    :: (HashAlgorithm algo, Bi a)
    => a -> AbstractHash algo b
unsafeAbstractHash = AbstractHash . Hash.hash . Bi.serialize'
```

As we are serialising a potentially big input (maxBound of `Word64`) and due to the fact CBOR is a var-len encoding, you might need more than 2 bytes overhead to encode the CBOR tag + the len of the BS + the actual BS. Does it makes sense?